diff --git a/Lib/Platform/POSIX/ThreadPOSIX.cpp b/Lib/Platform/POSIX/ThreadPOSIX.cpp
index b9aedc42..ae1ebbb7 100644
--- a/Lib/Platform/POSIX/ThreadPOSIX.cpp
+++ b/Lib/Platform/POSIX/ThreadPOSIX.cpp
@@ -60,7 +60,7 @@ struct CreateThreadArgs
 
 static constexpr Uptr sigAltStackNumBytes = 65536;
 
-#define ALLOCATE_SIGALTSTACK_ON_MAIN_STACK 1
+#define ALLOCATE_SIGALTSTACK_ON_MAIN_STACK 0
 
 static bool isAlignedLog2(void* pointer, Uptr alignmentLog2)
 {
